XPlat/Shopify/iOS/source/Shopify.iOS/ApiDefinition.cs

```diff
@@ -0,0 +1,1964 @@
+		string LastDigits { get; set; }
+
+		[Export ("expiryYear", ArgumentSemantic.Copy)]
+		NSNumber ExpiryYear { get; set; }
+
+		[Export ("expiryMonth", ArgumentSemantic.Copy)]
+		NSNumber ExpiryMonth { get; set; }
+
+	}
+
+	interface INavigationControllerDelegate
+	{
+
+	}
+
+	[Protocol, Model]
+	[BaseType (typeof(NSObject), Name = "BUYNavigationControllerDelegate")]
+	interface NavigationControllerDelegate
+	{
+		[Abstract]
+		[Export ("presentationControllerWillDismiss:")]
+		void PresentationControllerWillDismiss (UIPresentationController presentationController);
+
+		[Abstract]
+		[Export ("presentationControllerDidDismiss:")]
+		void PresentationControllerDidDismiss (UIPresentationController presentationController);
+
+	}
+
+	[BaseType (typeof(UINavigationController), Name = "BUYNavigationController")]
+	interface NavigationController : Themeable
+	{
+		[Export ("updateCloseButtonImageWithTintColor:duration:")]
+		void UpdateCloseButtonImage (bool tintColor, nfloat duration);
+
+		[NullAllowed, Export ("navigationDelegate", ArgumentSemantic.Weak)]
+		INavigationControllerDelegate NavigationDelegate { get; set; }
+	}
+
+	[BaseType (typeof(Object), Name = "BUYOption")]
+	interface Option
+	{
+		[Export ("name")]
+		string Name { get; }
+
+		[Export ("position", ArgumentSemantic.Strong)]
+		NSNumber Position { get; }
+
+		[Export ("productId", ArgumentSemantic.Copy)]
+		NSNumber ProductId { get; }
+
+	}
+
+	[BaseType (typeof(UIView), Name = "BUYVariantOptionBreadCrumbsView")]
+	interface VariantOptionBreadCrumbsView : Themeable
+	{
+		[Export ("hiddenConstraint", ArgumentSemantic.Strong)]
+		NSLayoutConstraint HiddenConstraint { get; set; }
+
+		[Export ("visibleConstraint", ArgumentSemantic.Strong)]
+		NSLayoutConstraint VisibleConstraint { get; set; }
+
+		[Export ("setSelectedBuyOptionValues:")]
+		void SetSelectedBuyOptionValues (string[] optionValuesNames);
+
+	}
+
+	[BaseType (typeof(NavigationController), Name = "BUYOptionSelectionNavigationController")]
+	interface OptionSelectionNavigationController : Themeable
+	{
+		[Export ("dismissWithCancelAnimation")]
+		bool DismissWithCancelAnimation { get; set; }
+
+		[Export ("breadsCrumbsView", ArgumentSemantic.Strong)]
+		VariantOptionBreadCrumbsView BreadsCrumbsView { get; set; }
+
+	}
+
+	interface IOptionSelectionDelegate
+	{
+
+	}
+
+	[Protocol, Model]
+	[BaseType (typeof(NSObject), Name = "BUYOptionSelectionDelegate")]
+	interface OptionSelectionDelegate
+	{
+		[Abstract]
+		[Export ("optionSelectionController:didSelectOptionValue:")]
+		void OptionSelectionController (OptionSelectionViewController controller, OptionValue optionValue);
+
+		[Abstract]
+		[Export ("optionSelectionControllerDidBackOutOfChoosingOption:")]
+		void OptionSelectionControllerDidBackOutOfChoosingOption (OptionSelectionViewController controller);
+
+	}
+
+	[BaseType (typeof(UITableViewController), Name = "BUYOptionSelectionViewController")]
+	interface OptionSelectionViewController : Themeable
```

Comment:
I would suggest to add [DisableDefaultCtor] attribute because OptionValue and FilteredProductVariantsForSelectionOption parameters cannot be set after instantiating.